pkg/xds/auth/k8s/authenticator.go

```diff
@@ -57,22 +44,23 @@ func (k *kubeAuthenticator) Authenticate(ctx context.Context, resource model.Res
 	if err != nil {
 		return err
 	}
-
-	k.authenticated.Put(credential, struct{}{})
 	return nil
 }
 
-func (k *kubeAuthenticator) verifyToken(ctx context.Context, credential auth.Credential, proxyNamespace, serviceAccountName string) error {
+func (k *kubeAuthenticator) verifyToken(ctx context.Context, credential auth.Credential, proxyNamespace, serviceAccountName, resourceName string) error {
 	tokenReview := &kube_auth.TokenReview{
 		Spec: kube_auth.TokenReviewSpec{
 			Token: credential,
 		},
 	}
+
 	if err := k.client.Create(ctx, tokenReview); err != nil {
-		return errors.Wrap(err, "call to TokenReview API failed")
+		log.Error(err, "fail to call Kubernetes API Server to verify dataplane token", "proxy", resourceName, "serviceAccountName", serviceAccountName)
+		return errors.New("call to TokenReview API failed")
 	}
 	if !tokenReview.Status.Authenticated {
-		return errors.Errorf("token doesn't belong to a valid user")
+		log.Info("[WARNING] fail to verify dataplane token", "error", tokenReview.Status.Error)
```

Comment:
I'd actually go with:

```
l = logger.With("resourceName", resourceName, "serviceAccountName", serviceAccountName)
l.Debug("dataplane token verification failed")
if tokenReview.Status.Error != "" {
  l.Error("dataplane token verification failed", "error", tokenReview.Status.Error)
  return errors.Errorf("token verification failed with an error")
}
return errors.Errorf("service account token failed authentication")
```

Someone reading the dataplane log should be able to make a difference between a bad token or not succeeding auth no?